components/cam/cime_config/buildnml

```diff
@@ -1,225 +1,245 @@
+        # and how it is specified internally
+
+        if atm_grid == 'T31':
+            atm_grid = "48x96"
+        elif atm_grid == 'T42':
+            atm_grid = "64x128"
+        elif atm_grid == 'T85':
+            atm_grid = "128x256"
+        elif atm_grid == 'T341':
+            atm_grid = "512x1024"
+
+        # Some settings for single column mode.
+        scm = ""
+        if pts_mode:
+            scm = "-scam -nosmp"
+
+        spmd = '-spmd'
+        if mpilib == 'mpi-serial':
+            spmd = "-nospmd"
+
+        smp = '-smp'
+        if nthrds_atm == 1 and not force_build_smp:
+            smp = "-nosmp"
+
+        # The ocean component setting is only used by CAM to do attribute matching for
+        # setting default tuning parameter values.  In SOM mode we want to use the same
+        # tunings as the fully coupled B compset, so set the ocean component to pop in
+        # that case.
+        ocn = comp_ocn
+        if ocn == "docn":
+            docn_mode = case.get_value("DOCN_MODE")
+            if docn_mode == "som":
+                ocn = "pop"
+
+        comp = ""
+        if comp_interface == "mct":
+            comp = "mct"
+        elif comp_interface == "esmf":
+            comp = "esmf"
+
+        cam_lib_dirs = ""
+        if "cosp" in cam_config_opts:
+            cam_lib_dirs = "-cosp_libdir {}/atm/obj/cosp".format(exeroot)
+
+        # level information for CAM is part of the atm grid name - and must be stripped out
+        nlev=""
+        regex_ = re.compile("(.+)L(.+)")
+        match_ = regex_.match(atm_grid)
+        if match_:
+            atm_grid = match_.groups()[0]
+            nlev = "-nlev " + match_.groups()[1]
+
+        # specify target OS name
+        os_opt = ""
+        if os_ == 'bgq':
+            os_opt = "-target_os bgq"
+
+        # TODO: cam/bld/configure needs to be converted to python
+        run_cmd_no_fail("{} -s -ccsm_seq -ice none -ocn {} -comp_intf {} {} -spmd {} -smp {} -dyn {} -dyn_target {} -res {} {} {} {} {}".format(os.path.join(srcroot, "components/cam/bld/configure"), ocn, comp, scm, spmd, smp, cam_dycore, cam_target, atm_grid, nlev, cam_lib_dirs, cam_config_opts, os_opt))
+
+    else:
+
+        # Verify that we have a config_cache file.
+        expect(os.path.exists(camconf_dir, "config_cache.xml"), "Missing cam cache file")
+
+    #--------------------------------------------------------------------
+    # Invoke cam build-namelist - output will go in $CASEBUILD/camconf
+    #--------------------------------------------------------------------
+
+    ignore = "-ignore_ic_date"
+    if "-01-01" in run_startdate or "-09-01" in run_startdate:
+        ignore = "-ignore_ic_year"
+
+    if cam_nml_use_case not in [None, "UNSET"]:
+        usecase = "-use_case " + cam_nml_use_case
+    else:
+        usecase = " "
+
+    inst_string = ""
+    for inst_counter in xrange(1, ninst_atm + 1):
```

Comment:
Python 3 compatibility for `xrange`: http://python-future.org/compatible_idioms.html#xrange
Or just use `range`

Reply:
Since this is E3SM, not CIME, I don't think we need to worry about py3 for now. @rljacob , what do you think?

Reply:
That's fair. I just didn't see anything else obvious (though looking again I see potential division issues) blocking python3 support so thought I'd mention it